Cython/Compiler/ExprNodes.py

```diff
@@ -13563,7 +13555,83 @@ def generate_result_code(self, code):
             code.error_goto_if_null(self.result(), self.pos)))
         code.put_gotref(self.result())
 
+class AnnotationNode(ExprNode):
+    # Deals with the two possible uses of an annotation.
+    # 1. The post PEP-563 use where an annotation is stored
+    #  as a string
+    # 2. The Cython use where the annotation can indicate an
+    #  object type
+    #
+    # doesn't handle the pre PEP-563 version where the
+    # annotation is evaluated into a Python Object
+
+    subexprs = []
+    def __init__(self, pos, expr, string=None):
+        """string is expected to already be a StringNode or None"""
+        ExprNode.__init__(self, pos)
+        if string is None:
+            # import doesn't work at top of file?
+            from .AutoDocTransforms import AnnotationWriter
+            string = StringEncoding.EncodedString(AnnotationWriter().write(expr))
+            string = StringNode(pos, unicode_value=string, value=string.as_utf8_string())
+        self.string = string
+        self.expr = expr
+
+    # def analyse_types doesn't exist because nothing needs doing
+
+    def coerce_to_pyobject(self, env):
+        return self.string
```

Comment:
This is dangerous. Calling `coerce_to_pyobject()` is intended to be semantically like `coerce_to(py_object_type)`. Here, both would do entirely different things.